src/pydata_sphinx_theme/theme/pydata_sphinx_theme/layout.html

```diff
@@ -2,13 +2,17 @@
 {%- import "static/webpack-macros.html" as _webpack with context %}
 
 {%- block css %}
+    <script>
+      document.documentElement.dataset.mode = localStorage.getItem("mode") || "{{ default_mode }}";
+      document.documentElement.dataset.theme = localStorage.getItem("theme") || "light"
+    </script>
+    {{ _webpack.head_js_preload() }}
```

Comment:
that was just me trying to avoid to write the script directly in the `layout.html`. It never worked so I extracted the 2 first lines from `head_js_preload()` to avoid flickering. 
If you want I can move back this line where it was at the end of the `css` block.